mods/default/nodes.lua

```diff
@@ -2836,10 +2840,109 @@ minetest.register_node("default:meselamp", {
 })
 
 minetest.register_node("default:mese_post_light", {
-	description = S("Mese Post Light"),
-	tiles = {"default_mese_post_light_top.png", "default_mese_post_light_top.png",
-		"default_mese_post_light_side_dark.png", "default_mese_post_light_side_dark.png",
-		"default_mese_post_light_side.png", "default_mese_post_light_side.png"},
+	description = S("Apple Wood Mese Post Light"),
+	tiles = {"default_wood.png", "default_wood.png",
+		"default_wood.png^default_mese_post_light_side_dark.png",
+		"default_wood.png^default_mese_post_light_side_dark.png",
+		"default_wood.png^default_mese_post_light_side.png",
+		"default_wood.png^default_mese_post_light_side.png"
+	},
+	wield_image = "default_mese_post_light_side.png",
+	drawtype = "nodebox",
+	node_box = {
+		type = "fixed",
+		fixed = {
+			{-2 / 16, -8 / 16, -2 / 16, 2 / 16, 8 / 16, 2 / 16},
+		},
+	},
+	paramtype = "light",
+	light_source = default.LIGHT_MAX,
+	sunlight_propagates = true,
+	is_ground_content = false,
+	groups = {choppy = 2, oddly_breakable_by_hand = 2, flammable = 2},
+	sounds = default.node_sound_wood_defaults(),
+})
+
+minetest.register_node("default:mese_post_light_acaciawood", {
+	description = S("Acacia Wood Mese Post Light"),
+	tiles = {"default_acacia_wood.png", "default_acacia_wood.png",
+		"default_acacia_wood.png^default_mese_post_light_side_dark.png",
+		"default_acacia_wood.png^default_mese_post_light_side_dark.png",
+		"default_acacia_wood.png^default_mese_post_light_side.png",
+		"default_acacia_wood.png^default_mese_post_light_side.png"
+	},
+	wield_image = "default_mese_post_light_side.png",
```

Comment:
You should probably use different wield images.

Reply:
This is gonna be a problem. The overlay texture stuff doesn't work well here.

Reply:
Hmm, is there a way to autogenerate a wield model or something?

Reply:
yes... by using `.. name ..` and then creating an alpha overlay image, as fences do

Reply:
It's either that or you'll have to make new wield textures for each

Reply:
Ugghhh. Just when I thought I had an efficient way of solving it xD